compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1967,7 +1972,11 @@ private BType visitCollectionType(BLangInputClause bLangInputClause, BType colle
                 if (tupleType.restType != null) {
                     tupleTypes.add(tupleType.restType);
                 }
-                return tupleTypes.size() == 1 ?
+                int tupleTypesSize = tupleTypes.size();
+                if (tupleTypesSize == 0) {
+                    return symTable.neverType;
+                }
+                return tupleTypesSize == 1 ?
```

Comment:
Isn't this block pretty much a duplicate of that for foreach? For now, shall we extract at lest the tuple bit to a separate method?

Reply:
Updated 376dc2f1f9bf11df4cd92cf83d6edff3cba48765